### ISSUE TITLE
Error while searching, and some things i noticed.

### ISSUE THREAD

**Issue by rubendmribeiro** [UnidentifiedX/YTSearch.NET#6]
I found this library recently and I'm certainly going to use it a lot as I started making a music app.
However, I would like a way to limit the number of results to be implemented in "await client.SearchYoutubeVideoAsync", or to create an offset so that as the user scrolls, it is possible to obtain a more fluid listing. 
The limitation also serves to avoid causing blockages that take a long time.
Also the search is not working i tried to search for "beautiful girls" and it pop up the following error.

![image](https://github.com/user-attachments/assets/7b180a68-60d2-4eb8-806d-11e21a2fdd41)

**Comment by UnidentifiedX** [comment]
Hey @kelputoo! I'll look into this as soon as possible, but I'm currently in school and can't work on this project right now.

When I originally started this project I tried to figure out a way to limit listings and also get more listings past listing 20. I couldn't figure out a way to do this back then, so I didn't implement it. I'll go have a look into this problem again, and will update my progress.

**Comment by rubendmribeiro** [comment]
Hi, it's okay. Take your time.
You're doing this with Web Scraping?

**Comment by UnidentifiedX** [comment]
@kelputoo yup, because back then when I started this project this seemed to be the only way to scrape video data without using an API key

**Comment by rubendmribeiro** [comment]
Yes, probably that's the only way.

**Comment by rubendmribeiro** [comment]
I'll give you an advice, if you want to invest time on this, try to make a better version of: "https://github.com/Tyrrrz/YoutubeExplode"
For example, with the limit search i said etc.

They don't have any limit, but u can apply by urself and it runs fast, for example, i did this way:

`` private async Task SearchYouTubeVideosAsync(string query)
 {
     // Clear the current items
     VideosCollectionView.ItemsSource = null;

     // Initialize YoutubeClient
     var youtube = new YoutubeClient();

     // Create an observable collection to hold the video results
     ObservableCollection<YouTubeVideo> Videos = new ObservableCollection<YouTubeVideo>();

     // Define the maximum number of results you want
     int maxResults = 25;
     int resultCount = 0;

     // Use await foreach to iterate over the search results
     await foreach (var video in youtube.Search.GetVideosAsync(query))
     {
         if (resultCount >= maxResults)
             break;

         var youtubeVideo = new YouTubeVideo
         {
             Title = video.Title,
             VideoUrl = $"https://www.youtube.com/watch?v={video.Id}",
             ThumbnailUrl = video.Thumbnails.FirstOrDefault()?.Url, // Safely accessing the first thumbnail
             Author = video.Author.ChannelTitle,
             Duration = video.Duration ?? TimeSpan.Zero // Handle nullable TimeSpan
         };

         Videos.Add(youtubeVideo);
         resultCount++;
     }

     // Set the items source for the CollectionView
     VideosCollectionView.ItemsSource = Videos;
 }`

**Comment by UnidentifiedX** [comment]
@kelputoo thanks for the repo! ill have a look at it :)

**Comment by UnidentifiedX** [comment]
@kelputoo Could you send more data about the crash you encountered? I am unable to replicate your error

**Comment by B0N3head** [comment]
Hey @kelputoo,

Firstly thanks for bringing the error to our attention 🙏

Could explain your current environment a little more and provide a full log of the error so we can try understand what exactly has gone wrong.

- What version of YTSearch.NET are you using?
- Did you download/install the package from this repo or from NuGet?
- Have you made any modifications to the library?
- I can see your using .net8, are you on a preview version or the latest 8.0.8?
- Have you attempted to replicate the error on any other .net versions?

The message box shown in the image perfectly cuts off as its about to log what portion of YTSearch.NET is causing the error. If you could provide another screenshot with the rest of the log or possibly log the error to console (to copy/paste).

I replicated your code in .net6 using YTSearch.Net v1.0.4 from NuGet and no errors were thrown. Would you be able to attempt the same in .net6 on your machine and provide a log (if it fails).

```c#
try
{
    var client = new YouTubeSearchClient();
    var results = await client.SearchYoutubeVideoAsync("beautiful girls");
    foreach (var result in results.Results)
    {
        Console.WriteLine($" {result.Title}\t{result.Author}\t{result.Length:mm\\:ss}");
    }
}
catch (Exception ex)
{
    Console.WriteLine(ex);
}
```

![image](https://github.com/user-attachments/assets/fb3b66de-e287-478c-90ae-8700b749fffd)


Cheers

**Comment by rubendmribeiro** [comment]
Hi, im currently not using the library.

However i will try to help the max. i can.
I was using the lastest .NET 8 Release Version
All the MAUI Nuget Packages were updated to the lastest RELEASE VERSION.
I've downloaded the package from Nuget and i was using the lastest version avaliable too.
**Have you attempted to replicate the error on any other .net versions?** No.
**Have you made any modifications to the library?** No.

**Comment by B0N3head** [comment]
> Hi, im currently not using the library.
> 
> However i will try to help the max. i can. I was using the lastest .NET 8 Release Version All the MAUI Nuget Packages were updated to the lastest RELEASE VERSION. I've downloaded the package from Nuget and i was using the lastest version avaliable too. **Have you attempted to replicate the error on any other .net versions?** No. **Have you made any modifications to the library?** No.

Yeah right, ok. 
That's super weird. 
I'll try replicate on my end using .net8 and see what happens 👌.

Did you pivot to using YoutubeExplode?
